src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlConnection.cs

```diff
@@ -119,7 +121,9 @@ public ResettableSqlStatement GetResettableStatement(string query)
         {
             if (!_queryToStatement.TryGetValue(query, out var statement))
             {
+#pragma warning disable CA2000 // Dispose objects before losing scope - dispose ownership transfer of rawStatement to SqlStatement instance.
```

Comment:
💭 This is actually a transfer of ownership to `ResettableSqlStatement` via an intermediate holder object `SqlStatement`. We should investigate improving the pattern by making `SqlStatement` disposable, which could then be detected by the analysis as a double-transfer `result`&rarr;`statement`&rarr;return.